sksparse/cholmod.pyx

```diff
@@ -764,10 +947,10 @@ cdef class Factor:
         """
 
         return self(sparse.eye(self._factor.n, self._factor.n,
-                               dtype=_np_dtype_for(self._factor.xtype),
+                               dtype=_np_dtype_for_data(self._factor.xtype),
                                format="csc"))
 
-def analyze(A, mode="auto", ordering_method="default"):
+def analyze(A, mode="auto", ordering_method="default", use_long=None):
```

Comment:
Do we really need to expose `use_long`?  Isn't it enough to just rely on the input type?